smithy-codegen-core/src/main/java/software/amazon/smithy/codegen/core/trace/Metadata.java

```diff
@@ -61,10 +61,10 @@ private ArtifactMetadata(Builder builder) {
      * @param value an ObjectNode that contains all children of the artifact tag in the trace file
      * @return ArtifactMetadata produced by deserializing the node.
      */
-    public static ArtifactMetadata fromNode(Node value) {
+    public static Metadata fromNode(Node value) {
```

Comment:
Can we call this `TraceMetadata` or something more descriptive? I fear that there are lots of classes called Metadata, and it would make autocomplete harder to find, and cause import collisions.